src/ethereum/shanghai/fork.py

```diff
@@ -559,8 +559,8 @@ def process_transaction(
 
     preaccessed_addresses = set()
     preaccessed_storage_keys = set()
+    preaccessed_addresses.add(env.coinbase)
     if isinstance(tx, (AccessListTransaction, FeeMarketTransaction)):
-        preaccessed_addresses.add(env.coinbase)
```

Comment:
This seems unrelated to the title of this pull request. Is it supposed to be here, and if so, should there be a test that differentiates these behaviours? Maybe a `LegacyTransaction` that tries to do a cold read from the coinbase?

Reply:
See https://github.com/ethereum/execution-specs/pull/761